tools/tests/test_osqueryd.py

```diff
@@ -66,5 +67,28 @@ def test_3_daemon_with_watchdog(self):
         # dies when the watcher goes away
         self.assertTrue(daemon.isDead(children[0]))
 
+    def test_4_daemon_sighup(self):
+        # A hangup signal should not do anything to the daemon.
+        daemon = self._run_daemon({
+            "disable_watchdog": True,
+        })
+        self.assertTrue(daemon.isAlive())
+
+        # Send a SIGHUP
+        os.kill(daemon.proc.pid, signal.SIGHUP)
+        self.assertTrue(daemon.isAlive())
+
+    def test_4_daemon_sigint(self):
+        # An interrupt signal will cause the daemon to stop.
+        daemon = self._run_daemon({
+            "disable_watchdog": True,
```

Comment:
There's a bit of an artifact/confound from the Python integration tests. The `python` process will subprocess and set the process group of the daemon/watcher as the pid of the python process. The worker's process group is the same, so sending a signal to the process group would also include `python`. We can modify the above two options to require no changes to the client and instead have python set a new group for the worker and watcher then test by sending a signal to the group instead of the daemon pid.